internal/plugin/instance_plugin.go

```diff
@@ -57,7 +60,9 @@ func (i *Instance) Process(_ context.Context, msg *bus.Message) {
 
 		instanceList := i.instanceService.GetInstances(newProcesses)
 		if len(instanceList) > 0 {
+			i.instancesMutex.Lock()
 			i.instances = instanceList
+			i.instancesMutex.Unlock()
```

Comment:
I'm curious why we are storing the instance list in the plugin, since it's not used anywhere?

Reply:
Its used in the unit tests

Reply:
Are we sure that we wanna add extra overhead (and locking) to production code just to make our code more testable?

Reply:
thats a far point. Looking at the code again I think I might be able to test the code a different way which doesn't require locks

Reply:
updated code to remove mutex